### ISSUE TITLE
TF 2.3, CUDA 11.1 , CUDNN 8.0.5, RT 7.2.1

### ISSUE THREAD

**Issue by JanSobus** [davidenunes/tensorflow-wheels#18]
Hey, hope you might be able to help me, because I've been banging my head against the wall the whole day.

I want a TF2.3 with CUDA 11.1, CUDNN 8.0.5 and RT 7.2.1., system is fresh ubuntu 20.04 
Tried these steps:
- Downloaded the newest versions (CUDA 11.1.1- 455.32 , Cudnn 8.0.5.398 and RT 7.2.1.6)
- After installing everything, trying to do the bazel build fails on cudart (it looks for version 11.1 and SONAME information in libcudart.so.11.1.74 is 11.0) .  Overriding the bazel compiler options with the method described here (https://towardsdatascience.com/how-to-compile-tensorflow-2-3-with-cuda-11-1-8cbecffcb8d3) makes the build complete but when I launch TF, it still pops error about being unable to load libcudart.so.11.1. Creating appropriately named softlink pointing at   libcudart.so.11.1.74 doesn't help, still the same error at TF import
- wiped the env and tried changing the internal SONAME in libcudart.so.11.1.74 from 11.0 to 11.1 using patchelf. Then the bazel build works without touching the compiler, however after installing the wheel, the same problem remains - TF still can't load libcudart.so.11.1 

Any ideas how to deal with that?
Thanks in advance!

**Comment by davidenunes** [comment]
I have [this new wheel ](https://github.com/davidenunes/tensorflow-wheels/releases/download/r2.4.rc4.cp38/tensorflow-2.4.0rc4-cp38-cp38-linux_x86_64.whl) for TF2.4rc4 with CUDA 11.1, cuDNN 8, TensorRT 7.2.1 

just tested it and it everything is working great. Assuming you have CUDA installed properly you should have no problem
the only thing is, you might need to upgrade your glibc. Since I'm on a rolling distro, I'm on 2.32

**Comment by JanSobus** [comment]
Thank for the quick reply! I might give it a try, would prefer to stay on 2.3 though, since all our codebase was developed in that version and I want to ensure backwards compatibility. Maybe it's a time to upgrade then ...

**Comment by davidenunes** [comment]
You can just check the breaking changes here: [https://github.com/tensorflow/tensorflow/releases/tag/v2.4.0-rc4](https://github.com/tensorflow/tensorflow/releases/tag/v2.4.0-rc4)

if any of the breaking changes are a deal breaker I can try to build for 2.3 later if I find the time.
I wanted to share this with you because it happens I was building it for myself. The only problem you might have 
is the fact that Ubuntu 20.04 comes with glibc 2.31 and it might complain trying to find the version I built this on 2.32

I'll also be building the 2.4 when it comes out.

**Comment by JanSobus** [comment]
I installed this 2.4 wheel in new env, it was complaining about lack of nccl, installed that. Then as you predicted glibc 2.3.2 was a problem, upgraded Ubuntu to 20.10 to solve that. And now it's again complaining about libcudart, the same error as I got, even though all the libs and links are at the correct spot. 
![image](https://user-images.githubusercontent.com/17416468/101230088-232a7200-36ef-11eb-8527-6b93fb8dfb8e.png)
